diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/generation/Generation.kt

```diff
@@ -47,6 +50,122 @@ private fun generateWarningNames() {
+                writer.writeln("""\hline""")
+                val columnNames = Regex("""[A-Za-z ]*""")
+                        .findAll(line)
+                        .filter { it.value.isNotEmpty() }
+                        .map { it.value.trim() }
+                        .toList()
+                writer.write(columnNames.joinToString(separator = "&"))
+                writer.writeln("""\\""")
+                writer.writeln("\\hline")
+                iterator.next()
+                line = iterator.next()
+                while(iterator.hasNext() && line.trim().startsWith("|")) {
+                    writer.writeln(line
+                            .replace('|', '&')
+                            .drop(1)
+                            .dropLast(1)
+                            .plus("""\\""")
+                            .replace("_", "\\_")
+                            .replace(Regex("""\(#(.*)\)"""), ""))
+                    line = iterator.next()
+                }
+                writer.writeln("""\hline""")
+                writer.writeln("\\end{tabular}")
+                writer.writeln("\\end{center}")
+            } else {
+                var correctedString = findBoldOrItalicText(line, Regex("""\*\*([^*]+)\*\*"""), FindType.BOLD)
+                correctedString = findBoldOrItalicText(correctedString, Regex("""\*([A-Za-z ]+)\*"""), FindType.ITALIC)
+                correctedString = findBoldOrItalicText(correctedString, Regex("""`([^`]*)`"""), FindType.BACKTICKS)
+                correctedString = correctedString.replace(Regex("""\(#(.*)\)"""), "")
+                correctedString = correctedString.replace("#", "\\#")
+                correctedString = correctedString.replace("&", "\\&")
+                correctedString = correctedString.replace("_", "\\_")
+                writer.writeln(correctedString)
+            }
+        }
+    }
+
+    appendixFile.removeAll(appendixFile.subList(appendixFile.indexOf("\\section*{guide}"), appendixFile.lastIndex + 1))
+    appendixFile.addAll(tempFile.readLines())
+    File("wp/sections/appendix.tex").writeText(appendixFile.joinToString(separator = "\n"))
+}
+
+enum class FindType {
+    BOLD,
+    ITALIC,
+    BACKTICKS
+}
+
+private fun findBoldOrItalicText(line: String, regex: Regex, type: FindType) : String {
+    val allRegex = regex.findAll(line).map { it.value }.toMutableList()
+    var correctedLine = line.replace(regex, "R_EP_LA_CE_ME")
+    allRegex.forEachIndexed {index, value ->
+        when (type) {
+            FindType.BOLD -> allRegex[index] = "\\textbf{${value.replace("**", "")}}"
+            FindType.ITALIC -> allRegex[index] = "\\textit{${value.replace("*", "")}}"
+            FindType.BACKTICKS -> allRegex[index] = value // currently not in use
+        }
+    }
+    allRegex.forEach {
+        correctedLine = correctedLine.replaceFirst("R_EP_LA_CE_ME", it)
+    }
+    return correctedLine
+}
+
+private fun PrintWriter.writeln(text: String) {
```

Comment:
These methods are sepcific to latex, if I understand correctly? I think they should be extracted to some sort of `LatexUtils` to have an explicit usage scope.